test/e2e/storage/sanity.go

```diff
@@ -995,3 +999,34 @@ func WaitForPodsWithLabelRunningReady(c clientset.Interface, ns string, label la
 		})
 	return pods, err
 }
+
+// We register list of volumes before test, using out-of-band commands (i.e. not CSI API but direct host commands).
+// checkForLeftovers lists volumes again after test, and comparing shows possible leftovers.
+// We avoid complexity of tracking nodes separately, by combining all outputs into same string.
+// TODO: remove debug prints before merging
+func getVolumesList(numWorkers int) string {
+	var cmd string
+	var buf string
+	switch os.Getenv("TEST_DEVICEMODE") {
+	case "direct":
+		cmd = "sudo ndctl list"
+	case "lvm":
+		cmd = "sudo lvs --foreign --noheadings"
+	}
+	framework.Logf("getVolumesList-DEBUG: cmd:[%s]", cmd)
+	for worker := 1; worker <= numWorkers; worker++ {
+		sshcmd := fmt.Sprintf("%s/_work/%s/ssh.%d", os.Getenv("REPO_ROOT"), os.Getenv("CLUSTER"), worker)
+		ssh := exec.Command(sshcmd, cmd)
+		out, _ := ssh.CombinedOutput()
+		//framework.Logf("get VolumeList-DEBUG(%d): cmd:[%s] out:[%s]", worker, cmd, out)
+		buf += string(out)
+	}
+	framework.Logf("getVolumeList-DEBUG: combined:[%s]", buf)
+	return buf
+}
+
+func checkForLeftoverVolumes(numWorkers int, volBefore string) {
+	volNow := getVolumesList(numWorkers)
+	framework.Logf("checkForLeftoverVolumes-DEBUG: now:[%s] before:[%s]", volNow, volBefore)
+	Expect(volNow).To(Equal(volBefore))
```

Comment:
This results in:
```
 Expected

[2019-11-04T10:06:27.311Z]           <string>:   6a2130accf186ac16f0d4f336bd488c9a512513f ndbus0region0fsdax -wi-a----- 4.00m                                                    \n

[2019-11-04T10:06:27.311Z]       to equal

[2019-11-04T10:06:27.311Z]           <string>: 
```

Can you add a description to `To`, something like `.To(Equal(volBefore), "same volumes before and after the test"`)?

Comparing strings is okay, but if we have many active volumes finding the one that was left over may become harder. Does the output become nicer when comparing lists of volumes (i.e. split at `\n` in `getVolumesList`)? I don't know how well gomega handles this, but there are toolkits which then nicely highlight changes.